### ISSUE TITLE
Support for async function in `time` function

### ISSUE THREAD

**Issue by cottinisimone** [mcasper/dogstatsd-rs#39]
Would be cool to be able to track duration of async functions too in `time` function. What i imagine is something like:

```rust
  use dogstatsd::{Client, Options};
  use std::thread;
  use std::time::Duration;

  let client = Client::new(Options::default()).unwrap();
  client
    .time("timer", &["tag:time"], || async {
      thread::sleep(Duration::from_millis(200))
    })
    .await
    .unwrap_or_else(|e| println!("Encountered error: {}", e))
```

Thank you in advance

**Comment by mcasper** [comment]
Instead of making the lambda async, could you instead stick the timing call inside the async function? Or is the intention to include the async scheduling time in the metric?

e.g.
```rust
async fn foo() {
  client.time("timer", tags, ...)
}
```

**Comment by cottinisimone** [comment]
Hello and thank you for the fast answer :).

The problem is that anyway if the code you are running is part of an async chain you'll anyway need that anonymous function to be async. Imagine for example if you need to time a query or an external api call. If the code is async there's no way to time that function in this way using `time` function. Ofc you can use `timing` function using `Instant`, then `elapsed` and putting the took time, but in this way `time` will always be useless.

Wdyt?

**Comment by mcasper** [comment]
I'm fine with adding support for tracking async functions, I just don't have a lot of time to give to this library at the moment. If this is something that you'd be interested in implementing, I'd be happy to review a PR!

**Comment by mcasper** [comment]
@cottinisimone I just released all your changes in 0.8.0, thanks so much! 😄
